### ISSUE TITLE
Sync EFI partition conditions with the "other" world

### ISSUE THREAD

**Issue by frispete** [yast/yast-storage-ng#1194]
Hi,

while installing TW 20201223 on a brand new system (ASUS ROG STRIX B550-F, AMD Ryzen 9 3900X, WD Black 1TB SN850 NVMe), that should perform in `both` worlds, I noticed the partitioner complaining about a too small efi partition (<128G) with the wrong type, resulting in a failed `shim-install` call later on. Luckily, I was able to fix this manually, but that shouldn't be necessary for the common case.

Still interested? Here are the details: The usual procedure is: install Windows 10 (Pro), get it to behave, and install TW into a resized primary Win partition. The system parts are pretty current (besides the AMD cpu from the last generation). Consequently, Windows didn't install, unless an 20H2 image was used (created with the latest MediaCreationTool).

I mention this in order to show, that the Microsoft part is current as well. Unfortunately, it seems to be using some default sizes, that YaST doesn't deal with properly, and screwing the OOB experience greatly.

Here's the final layout:
```
Disk /dev/nvme0n1: 931.53 GiB, 1000204886016 bytes, 1953525168 sectors
Disk model: WDS100T1X0E-00AFY0                      
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: FCAE70F3-XXXX-YYYY-ZZZZ-serial obscured

Device             Start        End    Sectors   Size Type
/dev/nvme0n1p1      2048     206847     204800   100M EFI System
/dev/nvme0n1p2    206848     239615      32768    16M Microsoft reserved
/dev/nvme0n1p3    239616  773332991  773093376 368.7G Microsoft basic data
/dev/nvme0n1p4 773332992 1953525134 1180192143 562.8G Linux filesystem
```

If Microsoft thinks, 100M is enough for the EFI partition, we should cope with that fact.
It looks, like things are well after manual intervention:
```
$ df  -h
/dev/nvme0n1p4         563G   55G  509G  10% /
/dev/nvme0n1p1          96M   31M   66M  32% /boot/efi
```

**Comment by dgdavid** [comment]
Thanks a lot for the detailed report @frispete!

Let's wait for @ancorgs insights after his short Christmas break.

**Comment by frispete** [comment]
My pleasure, @dgdavid, and sure, let's wait for @ancorgs for further discussions.

I forgot to mention, that the filesystem type of `/dev/nvme0n1p1` is fine:
```
/dev/nvme0n1p1        vfat       96M   31M   66M  32% /boot/efi
```
so, it's mostly a matter of reducing the minimal accepted size for the `EFI` partition from my humble POV...

**Comment by ancorgs** [comment]
> so, it's mostly a matter of reducing the minimal accepted size for the `EFI` partition from my humble POV...

Yes... and not. It's a matter of reducing (maybe to zero) the minimal accepted size for the ESP, but only when we are re-using an already existing partition and/or when the Expert Partitioner is used.
 

In my opinion, it's not a matter of reducing the minimal size used to propose a brand new ESP (it can be tweaked, but that's not the point here) but a matter 


Currently the minimal of 128 MiB is used in three situations:

- When the automatic partitioning suggests to create a new ESP because there is none (eg. empty disk). The exact number may be adjusted, but I still think 128 MiB is quite acceptable.
- When the automatic partitioning is deciding whether a pre-existing ESP should be reused instead of proposing a brand new one. That's the scenario described here. In my opinion, it's an error to use the same minimal here. That minimal size should not be a criterion to discard (and thus ignore) an ESP partition that looks fine otherwise.
- In the Expert Partitioner to check whether the user has created all the partitions needed for booting. Again, I don't think the same minimal should be used here. If the user used the Expert Partitioner to create a partition with the correct type and in the correct mount point... we can trust the size is fine.

**Comment by frispete** [comment]
> I will elaborate.

Cool, tried that myself, but failed.

> Currently the minimal of 128 MiB is used in three situations:
> 
> When the automatic partitioning suggests to create a new ESP because there is none (eg. empty disk). The exact number may be adjusted, but I still think 128 MiB is quite acceptable.

No, 128 MiB is fine. There's a comment in the code, that explains the rationals. (100 MiB MS and 28 MiB for us will provide enough headspace for some time)

> When the automatic partitioning is deciding whether a pre-existing ESP should be reused instead of proposing a brand new one. That's the scenario described here. In my opinion, it's an error to use the same minimal here. That minimal size should not be a criterion to discard (and thus ignore) an ESP partition that looks fine otherwise.

Exactly my point, but couldn't locate that logic in code.

> In the Expert Partitioner to check whether the user has created all the partitions needed for booting. Again, I don't think the same minimal should be used here. If the user used the Expert Partitioner to create a partition with the correct type and in the correct mount point... we can trust the size is fine.

I would second this as well.

With such changes in place, it will make it a much more pleasant user experience to put our beloved OS on a MS precharged system for obvious reasons.

**Comment by ancorgs** [comment]
Fixed by #1197
